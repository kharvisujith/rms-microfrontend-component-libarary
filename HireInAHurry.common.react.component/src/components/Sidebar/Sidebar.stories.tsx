import React from "react";
import { Meta, StoryFn } from "@storybook/react";
import Sidebar from "./Sidebar";
import LogoutRoundedIcon from "@mui/icons-material/LogoutRounded";
import SettingsRoundedIcon from "@mui/icons-material/SettingsRounded";
import WorkIcon from "@mui/icons-material/Work";
import CorporateFareIcon from "@mui/icons-material/CorporateFare";
import SupervisorAccountIcon from "@mui/icons-material/SupervisorAccount";
import HomeIcon from "@mui/icons-material/Home";

export default {
  title: "Components/Sidebar",
  component: Sidebar,
} as Meta<typeof Sidebar>;

const Template: StoryFn<typeof Sidebar> = (args) => <Sidebar {...args} />;

export const Default = Template.bind({});
Default.args = {
  sidebarItems: [
    { icon: <HomeIcon fontSize="large" />, label: "HOME" },
    { icon: <SupervisorAccountIcon fontSize="large" />, label: "CANDIDATES" },
    { icon: <CorporateFareIcon fontSize="large" />, label: "ORGANIZATION" },
    { icon: <WorkIcon fontSize="large" />, label: "JOBS" },
    { icon: <SettingsRoundedIcon fontSize="large" />, label: "SETTINGS" },
    { icon: <LogoutRoundedIcon fontSize="large" />, label: "LOGOUT" },
  ],
  handleOnClick: (label) => {
    console.log("label", label);
  },
};
