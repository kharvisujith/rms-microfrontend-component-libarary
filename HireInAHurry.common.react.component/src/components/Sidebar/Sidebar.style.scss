@import "../../common/common.style.scss";

.common-sidebar {
  display: flex;
  flex-direction: inherit;
  height: 100vh;
  &-container {
    // background-color: #0f2953;
    background-color: $sidebar-background-color;
    color: $sidebar-color;
    // color: #ffffff;
    width: fit-content;
    border-radius: 0 1rem 0 0;
    padding: 2rem 1.25rem;
    transform: translateX(0);
    transition: transform 1s, opacity 1s;

    &-expanded {
      padding: 2rem 1.5rem;
      transform: translateX(0);
      transition: transform 1s, opacity 0.5s;
      width: 11rem;
    }

    &-collapsed {
      padding: 2rem 0.625rem;
      transform: translateX(-5%);
      transition: transform 1s, opacity 0.5s;
    }

    &-toggle {
      &-expanded {
        display: flex;
        justify-content: flex-end;
      }
      &-icon {
        cursor: pointer;
      }

      &-hamburger-menu {
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        width: 1.875rem;
        background-color: #263e64;
        border-radius: 3.125rem;
        padding: 0.9375rem 0.625rem 0.75rem 0.625rem;
        cursor: pointer;
      }

      &-hamburger-menu-line {
        width: 100%;
        height: 0.1875rem;
        background-color: $sidebar-color;
        margin-bottom: 0.3125rem;
        transition: transform 0.5s, opacity 0.5s;
      }

      &-open .common-sidebar-container-toggle-hamburger-menu-line:nth-child(1) {
        transform: translateY(0.75rem) translateX(-0.5625rem) rotate(45deg);
        width: 50%;
      }

      &-open .common-sidebar-container-toggle-hamburger-menu-line:nth-child(2) {
        transform: translateY(0rem) rotate(180deg);
        width: 90%;
      }

      &-open .common-sidebar-container-toggle-hamburger-menu-line:nth-child(3) {
        transform: translateY(-0.75rem) translateX(-0.5625rem) rotate(-45deg);
        width: 50%;
      }
    }

    &-icon {
      display: flex;
      flex-direction: column;
      align-items: center;
      height: 92%; //changed from 100%
      justify-content: space-around;
      cursor: pointer;
      &-top {
        display: flex;
        flex-direction: column;
        height: 40%;
        justify-content: space-between;
        &-wrap {
          display: flex;
          align-items: center;
          cursor: pointer;
          .sidebar-icon-expanded {
            margin-left: 0.3125rem;
          }
          &-inner {
            width: 2rem in !important;
            cursor: pointer;
          }
        }
        &-wrap.active {
          background-color: #263e64;
          padding: 0.3125rem 0.625rem;
          border-radius: 0 0.625rem 0.625rem 0;
        }
      }

      &-bottom {
        display: flex;
        flex-direction: column;
        height: 20%;
        justify-content: space-between;
        //align-items: center;
        &-wrap {
          display: flex;
          align-items: center;
          cursor: pointer;
          .sidebar-icon-expanded {
            margin-left: 0.3125rem;
          }
          &-inner {
            width: 2rem in !important;
            cursor: pointer;
          }
        }
        &-wrap.active {
          background-color: #263e64;
          padding: 0.3125rem 0.625rem;
          border-radius: 0 0.625rem 0.625rem 0;
        }
      }
      &-bottom.expanded {
        height: 15%;
      }
    }

    &-icon.expanded {
      align-items: flex-start;
    }

    // .css-tzssek-MuiSvgIcon-root {
    //   width: 2rem !important;
    //   height: 2rem !important;
    // }
  }

  &-content {
    margin: 0 1rem;
    width: 100%;
  }
}
