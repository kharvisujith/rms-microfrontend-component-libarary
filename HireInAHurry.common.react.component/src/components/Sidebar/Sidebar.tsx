import React, { useState } from "react";
import "./Sidebar.style.scss";
import { Icon } from "@mui/material";

interface sidebarItems {
  icon: JSX.Element;
  label: string;
}

export interface SidebarProp {
  sidebarItems: sidebarItems[];
  handleOnClick: (label: string) => void;
}

const Sidebar: React.FC<SidebarProp> = ({
  sidebarItems,
  handleOnClick,
}: SidebarProp) => {
  const [activeContent, setActiveContent] = useState("HOME");
  const [sidebarExpanded, setSidebarExpanded] = useState(false);

  const activeClass = (label: string) =>
    activeContent === label ? "active" : "";

  const handleActiveLabel = (label: string) => {
    setActiveContent(label);
    handleOnClick(label);
  };

  const toggleSidebar = () => {
    setSidebarExpanded(!sidebarExpanded);
  };

  return (
    <div className={`common-sidebar`}>
      <div
        className={`common-sidebar-container ${
          sidebarExpanded
            ? "common-sidebar-container-expanded"
            : "common-sidebar-container-collapsed"
        }`}
      >
        <div
          className={`common-sidebar-container-toggle ${
            sidebarExpanded ? "common-sidebar-container-toggle-expanded" : ""
          }`}
          onClick={toggleSidebar}
        >
          <div
            className={`common-sidebar-container-toggle-hamburger-menu ${
              sidebarExpanded ? "common-sidebar-container-toggle-open" : ""
            }`}
            onClick={toggleSidebar}
          >
            <div className="common-sidebar-container-toggle-hamburger-menu-line"></div>
            <div className="common-sidebar-container-toggle-hamburger-menu-line"></div>
            <div className="common-sidebar-container-toggle-hamburger-menu-line"></div>
          </div>
        </div>
        {!sidebarExpanded ? (
          <div className={`common-sidebar-container-icon`}>
            <div className={`common-sidebar-container-icon-top`}>
              {sidebarItems
                .slice(0, 4)
                .map(({ icon, label }, index: number) => (
                  <div
                    key={index}
                    className={`common-sidebar-icon ${activeClass(label)}`}
                    onClick={() => handleActiveLabel(label)}
                  >
                    <Icon
                      sx={{
                        width: "2rem",
                        height: "2rem",
                      }}
                    >
                      {icon}
                    </Icon>
                  </div>
                ))}
            </div>
            <div className="common-sidebar-container-icon-bottom">
              {sidebarItems.slice(4).map(({ icon, label }, index: number) => (
                <div
                  key={index}
                  className={`common-sidebar-icon ${activeClass(label)}`}
                  onClick={() => handleActiveLabel(label)}
                >
                  <Icon
                    sx={{
                      width: "2rem",
                      height: "2rem",
                    }}
                  >
                    {icon}
                  </Icon>
                </div>
              ))}
            </div>
          </div>
        ) : (
          <div className={`common-sidebar-container-icon expanded`}>
            <div className={`common-sidebar-container-icon-top`}>
              {sidebarItems
                .slice(0, 4)
                .map(({ icon, label }, index: number) => (
                  <div
                    key={index}
                    className={`common-sidebar-container-icon-top-wrap ${activeClass(
                      label
                    )}`}
                    onClick={() => handleActiveLabel(label)}
                  >
                    <div
                      key={label}
                      className="common-sidebar-container-icon-top-wrap-inner"
                    >
                      <Icon
                        sx={{
                          width: "2rem",
                          height: "2rem",
                          marginRight: "1rem",
                        }}
                      >
                        {icon}
                      </Icon>
                    </div>
                    <div className="common-sidebar-container-icon-top-wrap-inner-label">
                      {label}
                    </div>
                  </div>
                ))}
            </div>
            <div className="common-sidebar-container-icon-bottom expanded">
              {sidebarItems.slice(4).map(({ icon, label }, index: number) => (
                <div
                  key={index}
                  className={`common-sidebar-container-icon-bottom-wrap ${activeClass(
                    label
                  )}`}
                  onClick={() => handleActiveLabel(label)}
                >
                  <div
                    key={label}
                    className={`common-sidebar-container-icon-bottom-wrap-inner`}
                  >
                    <Icon
                      sx={{
                        width: "2rem",
                        height: "2rem",
                        marginRight: "1rem",
                      }}
                    >
                      {icon}
                    </Icon>
                  </div>
                  <div className="common-sidebar-container-icon-top-wrap-inner-label">
                    {label}
                  </div>
                </div>
              ))}
            </div>
          </div>
        )}
      </div>
    </div>
  );
};

export default Sidebar;
