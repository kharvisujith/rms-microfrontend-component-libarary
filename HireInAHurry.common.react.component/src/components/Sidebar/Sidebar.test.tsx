import "@testing-library/jest-dom";
import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Sidebar, { SidebarProp } from "./Sidebar";

describe("Sidebar", () => {
  const sidebarItems: SidebarProp["sidebarItems"] = [
    {
      icon: <svg />,
      label: "Home",
    },
    {
      icon: <svg />,
      label: "About",
    },
    {
      icon: <svg />,
      label: "Contact",
    },
  ];

  const handleOnClickMock = jest.fn();

  it("renders correctly", () => {
    const { container } = render(
      <Sidebar sidebarItems={sidebarItems} handleOnClick={handleOnClickMock} />
    );
    expect(container).toMatchSnapshot();
  });

  it("renders the sidebar with default props", () => {
    const { container } = render(
      <Sidebar sidebarItems={sidebarItems} handleOnClick={handleOnClickMock} />
    );

    expect(container.querySelector(".common-sidebar")).toBeInTheDocument();

    expect(container.querySelectorAll(".common-sidebar-icon")).toHaveLength(3);

    expect(
      container.querySelectorAll(".common-sidebar-icon")[1]
    ).toBeInTheDocument();
  });

  it("calls handleOnClick when a sidebar item is clicked", () => {
    const { container } = render(
      <Sidebar sidebarItems={sidebarItems} handleOnClick={handleOnClickMock} />
    );

    fireEvent.click(container.querySelectorAll(".common-sidebar-icon")[1]);

    expect(handleOnClickMock).toHaveBeenCalledWith("About");
  });

  it("expands the sidebar when the toggle button is clicked", () => {
    const { container } = render(
      <Sidebar sidebarItems={sidebarItems} handleOnClick={handleOnClickMock} />
    );

    const toggleElement = container.querySelector(".common-sidebar-container-toggle");
    if (toggleElement) {
      fireEvent.click(toggleElement);
    }

    expect(
      container.querySelector(".common-sidebar-container-expanded")
    ).toBeInTheDocument();
  });
});
