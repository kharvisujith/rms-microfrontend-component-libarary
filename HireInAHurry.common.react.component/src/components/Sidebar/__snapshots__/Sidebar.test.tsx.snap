// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correctly 1`] = `
<div>
  <div
    class="common-sidebar"
  >
    <div
      class="common-sidebar-container common-sidebar-container-collapsed"
    >
      <div
        class="common-sidebar-container-toggle "
      >
        <div
          class="common-sidebar-container-toggle-hamburger-menu "
        >
          <div
            class="common-sidebar-container-toggle-hamburger-menu-line"
          />
          <div
            class="common-sidebar-container-toggle-hamburger-menu-line"
          />
          <div
            class="common-sidebar-container-toggle-hamburger-menu-line"
          />
        </div>
      </div>
      <div
        class="common-sidebar-container-icon"
      >
        <div
          class="common-sidebar-container-icon-top"
        >
          <div
            class="common-sidebar-icon "
          >
            <span
              aria-hidden="true"
              class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-59v7gf-MuiIcon-root"
            >
              <svg />
            </span>
          </div>
          <div
            class="common-sidebar-icon "
          >
            <span
              aria-hidden="true"
              class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-59v7gf-MuiIcon-root"
            >
              <svg />
            </span>
          </div>
          <div
            class="common-sidebar-icon "
          >
            <span
              aria-hidden="true"
              class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-59v7gf-MuiIcon-root"
            >
              <svg />
            </span>
          </div>
        </div>
        <div
          class="common-sidebar-container-icon-bottom"
        />
      </div>
    </div>
  </div>
</div>
`;
